### PR TITLE
Add sorted()

### LINKED ISSUE #6: Add sorted

### PR DESCRIPTION
**Pull request by kelunik**
Closes #6.

### CODE REVIEW

File: src/Internal/FlatMapOperation.php [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+                    $iterable = ($this->flatMap)($value, $position);
+                    foreach ($iterable as $item) {
+                        if ($item === $stop) {
+                            return;
+                        }
+
+                        yield $item;
+                    }
+                }
+            })());
```
Comment (kelunik) [inline src/Internal/FlatMapOperation.php:58]: I've changed this to use `ConcurrentIterableIterator`

File: src/Pipeline.php [hunk 2]
```diff
@@ -429,11 +483,10 @@ public function takeWhile(\Closure $predicate): self
         $sequence = new Sequence;
         $taking = true;
 
-        return $this->process(
-            $this->concurrency,
+        return $this->flatMap(
             function (mixed $value, int $position) use ($sequence, $predicate, &$taking) {
                 if (!$taking) {
-                    return false;
+                    return [];
```
Comment (kelunik) [inline src/Pipeline.php:489]: Quite nice that psalm found this.

File: src/Internal/FlatMapOperation.php (hunk 1 above)
Comment (trowski) [reply src/Internal/FlatMapOperation.php:58]: Does this introduce any difference between concurrency > 1? Should each fiber below use `ConcurrentIterableIterator` too?

Comment (kelunik) [reply src/Internal/FlatMapOperation.php:58]: It makes it lazy, too, yes. We could use a slightly modified version for the concurrent version, yes.

**Review by trowski — APPROVED** [review]

**Pull request merged**